x/commitment/keeper/keeper.go

```diff
@@ -299,3 +309,82 @@ func (k Keeper) ProcessWithdrawValidatorCommission(ctx sdk.Context, delegator st
 
 	return nil
 }
+
+// Withdraw Token - USDC
+// Only withraw USDC from dexRevenue wallet
+func (k Keeper) ProcessWithdrawUSDC(ctx sdk.Context, creator string, denom string, amount sdk.Int) error {
```

Comment:
I thought USDC withdrawal is done on incentive module. Is there any specific reason processing this on Commitment module? I think commitment module is in over-charge of what it's doing.

Reply:
We need to deduct uncommittedtoken (usdc) amount in commitment module. As all rewards are stored in commitment module in the state of committed and uncommitted. We just deduct the amount of uncommitted usdc amount.

Reply:
If we prefer to store reward tokens in commitment module, it would be okay. Would like to hear from @cosmic-vagabond on storing reward tokens on commitment module or separate one.

Reply:
@jelysn accounting all rewards in commitment is our original design.

Reply:
@jelysn @kenta-elys that’s correct that align with the original design of using commitment as an input/output interface for tokens that includes rewards

Reply:
@jelysn 
We will have all rewards accounted in the form of committed & uncommitted state in commitment module.